clis/teliod/src/interface_configurator.rs

```diff
@@ -0,0 +1,120 @@
+use crate::TeliodError;
+use serde::Deserialize;
+use std::net::IpAddr;
+use std::process::Command;
+use tracing::{error, info};
+
+pub trait InterfaceConfigurator {
+    /// Configure the IP address and routes for a given interface
+    fn set_ip(&self, _adapter_name: &str, _ip_address: &IpAddr) -> Result<(), TeliodError> {
+        Ok(())
+    }
+
+    /// Helper function to execute a system command
+    fn execute(&self, command: &mut Command) -> Result<(), TeliodError> {
```

Comment:
This has nothing to do with the trait it is defined on, move it to a free function or a seperate trait if it has a need for specilization

Reply:
Good point